click_odoo_contrib/update.py

```diff
@@ -31,12 +32,19 @@ def _get_param(cr, key, default=None):
 
 
 def _set_param(cr, key, value):
-    cr.execute(
-        "INSERT INTO ir_config_parameter "
-        "(key, value) VALUES (%s, %s) "
-        "ON CONFLICT (key) DO UPDATE SET value=%s",
-        (key, value, value),
-    )
+    try:
+        with cr.savepoint():
+            cr.execute(
+                "INSERT INTO ir_config_parameter " "(key, value) VALUES (%s, %s)",
+                (key, value),
+            )
+    except IntegrityError:
+        cr.execute(
+            """UPDATE ir_config_parameter
+               SET value=%s
+               WHERE key=%s""",
+            (value, key),
+        )
```

Comment:
I tend to prefer "update where... returning id" and check the result: if empty, insert. Just a matter of taste.

Reply:
Good idea. In fact, I changed to use [`rowcount`](http://initd.org/psycopg/docs/cursor.html#cursor.rowcount), which seems even simpler.